Assets/MixedRealityToolkit/Inspectors/MixedRealityWireframeShaderGUI.cs

```diff
@@ -0,0 +1,210 @@
+            materialEditor.ShaderProperty(wireColor, Styles.wireColor);
+            materialEditor.ShaderProperty(wireThickness, Styles.wireThickness);
+
+            AdvancedOptions(materialEditor, material);
+
+            //base.OnGUI(materialEditor, props);
+        }
+
+        protected void FindProperties(MaterialProperty[] props)
+        {
+            renderingMode = FindProperty(BaseStyles.renderingModeName, props);
+            customRenderingMode = FindProperty(BaseStyles.customRenderingModeName, props);
+            sourceBlend = FindProperty(BaseStyles.sourceBlendName, props);
+            destinationBlend = FindProperty(BaseStyles.destinationBlendName, props);
+            blendOperation = FindProperty(BaseStyles.blendOperationName, props);
+            depthTest = FindProperty(BaseStyles.depthTestName, props);
+            depthWrite = FindProperty(BaseStyles.depthWriteName, props);
+            depthOffsetFactor = FindProperty(BaseStyles.depthOffsetFactorName, props);
+            depthOffsetUnits = FindProperty(BaseStyles.depthOffsetUnitsName, props);
+            colorWriteMask = FindProperty(BaseStyles.colorWriteMaskName, props);
+
+            cullMode = FindProperty("_CullMode", props);
+            renderQueueOverride = FindProperty("_RenderQueueOverride", props);
+
+            baseColor = FindProperty("_BaseColor", props);
+            wireColor = FindProperty("_WireColor", props);
+            wireThickness = FindProperty("_WireThickness", props);
+        }
+
+        public override void AssignNewShaderToMaterial(Material material, Shader oldShader, Shader newShader)
+        {
+            float? cullMode = GetFloatProperty(material, "_Cull");
+
+            base.AssignNewShaderToMaterial(material, oldShader, newShader);
+   
+            SetFloatProperty(material, null, "_CullMode", cullMode);
+
+            // Setup the rendering mode based on the old shader.
+            if (oldShader == null || !oldShader.name.Contains("Legacy Shaders/"))
+            {
+                SetupMaterialWithRenderingMode(material, (RenderingMode)material.GetFloat(BaseStyles.renderingModeName), CustomRenderingMode.Opaque, -1);
+            }
+            else
+            {
+                RenderingMode mode = RenderingMode.Opaque;
+
+                if (oldShader.name.Contains("/Transparent/Cutout/"))
+                {
+                    mode = RenderingMode.TransparentCutout;
+                }
+                else if (oldShader.name.Contains("/Transparent/"))
+                {
+                    mode = RenderingMode.Transparent;
+                }
+
+                material.SetFloat(BaseStyles.renderingModeName, (float)mode);
+
+                MaterialChanged(material);
+            }
+        }
+
+        protected void Initialize(Material material)
+        {
+            if (!initialized)
+            {
+                MaterialChanged(material);
+                initialized = true;
+            }
+        }
```

Comment:
It looks like this + the initialized state are also common between the two current shaders - if the goal is to make the base one a common set of stuff, maybe this would be worth moving in there as well. If your thoughts are that this might not be something interesting to all subclasses, then it's fine to also keep out. Really just trying to understand the standards/thoughts that go into what is being made common.

Reply:
@wiwei yea I am on the fence with this one. I didn't put it in base because it relies on the materialproperty for the given shader (protected MaterialProperty renderingMode;) So I would have to move all of those to the base class for both WireframeGUI and StandardGUI, which is doable....thinking about it more, I think it's worth it. For any shaders in the future, we should be able to define rendering modes, transparency, etc. These kind of properties are fairly universal whether it's a lit or unlit shader or whatever. If for whatever reason we need to create a new shaderGUI that doesn't have these, then we can easily insert that